utils/utils.h

```diff
@@ -121,15 +121,11 @@
             pthread_mutex_unlock(&x);                                \
           } while (0)
 
-# define ERROR(fmt, ...)                                             \
-         do {                                                        \
-           fprintf(stderr, "Error: " fmt " [at %s+%d :<%s>]\n",      \
-                   __VA_ARGS__, __FILE__, __LINE__, __FUNCTION__);   \
-          } while (0)
-
-# define  MSG(fmt, ...)                                              \
+# define  MSG(fd, fmt, ...)                                          \
           do {                                                       \
-            fprintf(stdout, fmt, __VA_ARGS__);                       \
+            if (fd < 0)                                              \
+              fd = stderr;                                           \
```

Comment:
Here I think it should be fd > 0, the 0 is the stdin.
Or here it should only allow stdout and stderr ?

Reply:
> Here I think it should be fd > 0

No

Usually fd < 0 are not valid, so route the output to stderr (fd = stderr)

If stderr, is supplied the output __VA_ARGS__ should be printed on stderr
if stdout, is supplied the output should be printed on stdout.
If fd < 0  is supplied, lets print the output to stderr.

For reference
```
STDIN_FILENO  
Standard input value stdin. Its value is 0.

STDOUT_FILENO  
Standard output value stdout. Its value is 1.

STDERR_FILENO  
Standard error value stderr. Its value is 2.
```

Did I miss something here ?

Reply:
I just wanted to make sure that you were aware that when the fd == 0, which is the stdin. Will this make sense here ? I am okay with this.

Reply:
Ah! so you are saying line 126 should rather be 
```
if (fd <= 0) {
  fd = stderr;
}
```

I'm okay with this too. Let me know, if this is what you have request for.

Thanks @lxbsz

Reply:
Yeah, this was what I meant :)

Reply:
Cool :-)
Will refresh the PR with this change very soon.

Thanks @lxbsz

Reply:
This mostly needs taking care? Or is it already done!

Reply:
@amarts What are we looking for ?
According to us, we are done with this.

Reply:
From the comment, I didn't realize if it was done! Hence the comment. Patch looks OK for me.

Reply:
Cool.